APC control console fetches APCs from station levels, not just its own zlevel

Sealed101 opened the pull request:
## About The Pull Request

Title; also fixed a runtime in `restore_comp` where no user was provided to log the message. Also changed some var names and dmdocs the place

## Why It's Good For The Game

Fixes #72868

## Changelog

:cl:
fix: fixed APC control console found in CE office only accessing APCs on its z-level
fix: fixed restoring an emagged APC control console potentially leaving the console in a forever state of restoring
/:cl:

Sealed101 committed:
check for station z instead of console z

Sealed101 committed:
fix log restoring runtime

Sealed101 committed:
dmdoc and var names

Sealed101 committed:
active_apc comment fix

Ben10Omintrix commented on code/game/machinery/computer/apc_control.dm:
```diff
@@ -6,16 +6,17 @@
 	light_color = LIGHT_COLOR_DIM_YELLOW
-	var/obj/machinery/power/apc/active_apc //The APC we're using right now
+	///The APC we're remotely connected to right now
+	var/obj/machinery/power/apc/active_apc
```
bit unrelated to ur fix but do we clean up after this ref anywhere?

Sealed101 replied:
handled the apc being destroyed a bit better, hopefully

Ben10Omintrix replied:
nice job, should be enough

Ben10Omintrix commented on code/game/machinery/computer/apc_control.dm:
```diff
@@ -125,18 +141,18 @@
 				return
-			var/mob/living/operator = usr
-			if(!istype(operator))
+			var/mob/living/user = usr
```
```suggestion
			var/mob/living/user = ui.user
```

Ben10Omintrix commented on code/game/machinery/computer/apc_control.dm:
```diff
@@ -125,18 +141,18 @@
-			if(ID && istype(ID))
-				if(check_access(ID))
+			var/obj/item/card/id/user_id_card = user.get_idcard(TRUE)
+			if(user_id_card && istype(user_id_card))
```
```suggestion
			if(istype(user_id_card))
```
istype will return false if its null anyway

Sealed101 committed:
more fixes and qdel cleanup
apc frame unweld ballon alert fix
apc destruction won't cause it to try to speak while disconnecting remote user
away computers won't connect to station
prevents apc disconnection from firing on null apc when console's stat changes

Ben10Omintrix commented on code/game/machinery/computer/apc_control.dm:
```diff
@@ -153,12 +183,12 @@
 		if("restore-console")
 			restoring = TRUE
-			addtimer(CALLBACK(src, PROC_REF(restore_comp)), rand(3,5) * 9 SECONDS)
+			addtimer(CALLBACK(src, PROC_REF(restore_comp), usr), rand(3,5) * 9 SECONDS)
```
```suggestion
			addtimer(CALLBACK(src, PROC_REF(restore_comp), ui.user), rand(3,5) * 9 SECONDS)
```
not to be a pain but id appreciate these changes too. (alternatively, since we're using `ui.user` alot, u could cache it into a /mob/living reference before entering the switch statement and use that instead, but it doesnt matter much)

Sealed101 replied:
it's best to be a pain than leave these bits to the next guy that may not come

Ben10Omintrix commented on code/game/machinery/computer/apc_control.dm:
```diff
@@ -153,12 +183,12 @@
-			var/obj/machinery/power/apc/APC = locate(ref) in SSmachines.get_machines_by_type_and_subtypes(/obj/machinery/power/apc)
-			connect_apc(APC, usr)
+			var/obj/machinery/power/apc/remote_target = locate(ref) in SSmachines.get_machines_by_type_and_subtypes(/obj/machinery/power/apc)
+			connect_apc(remote_target, usr)
```
```suggestion
			connect_apc(remote_target, ui.user)
```

Ben10Omintrix commented on code/game/machinery/computer/apc_control.dm:
```diff
@@ -196,10 +226,10 @@
-			var/setTo = target.operating ? "On" : "Off"
-			log_activity("Turned APC [target.area.name]'s breaker [setTo]")
+			var/obj/machinery/power/apc/breaker_target = locate(ref) in SSmachines.get_machines_by_type_and_subtypes(/obj/machinery/power/apc)
+			breaker_target.toggle_breaker(usr)
```
```suggestion
			breaker_target.toggle_breaker(ui.user)
```

Sealed101 committed:
usr begone

Sealed101 committed:
Merge branch 'master' into apc_control_zlevel_fix
